core/client-internal/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -72,6 +72,9 @@ public AlluxioBlockStore getAluxioBlockStore() {
    * {@link ClientContext}.
    */
   public void reset() {
+    if (null != mFileSystemMasterClientPool) {
```

Comment:
Instead of reusing the same pool, wouldn't a cleaner solution is to close the connection for each client in the pool? 
And when the master address changes, you'll need to reinitialize the client pool, right?